pkg/registry/autoscaling/horizontalpodautoscaler/strategy.go

```diff
@@ -67,16 +65,7 @@ func (autoscalerStrategy) GetResetFields() map[fieldpath.APIVersion]*fieldpath.S
 }
 
 // PrepareForCreate clears fields that are not allowed to be set by end users on creation.
-func (autoscalerStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
-	newHPA := obj.(*autoscaling.HorizontalPodAutoscaler)
-
-	// create cannot set status
-	newHPA.Status = autoscaling.HorizontalPodAutoscalerStatus{}
-
-	if !utilfeature.DefaultFeatureGate.Enabled(features.HPAContainerMetrics) {
-		dropContainerMetricSources(newHPA.Spec.Metrics)
-	}
-}
+func (autoscalerStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {}
```

Comment:
Am I missing some context as to why we're wiping this entire function instead of just the feature gate check?

Same holds true for `PrepareForUpdate` below.

Reply:
You're right. I overlooked it prevents the operation from changing status.